src/test/java/uk/gov/hmcts/reform/userprofileapi/service/impl/UserProfileUpdatorTest.java

```diff
@@ -165,20 +167,51 @@ public void addRoles_InternalServerError() throws Exception {
         ObjectMapper mapper = new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
         String body = mapper.writeValueAsString(userProfileResponse);
 
+        Response responseMock = mock(Response.class);
+        when(responseMock.status()).thenReturn(500);
+
+        when(userProfileRepositoryMock.findByIdamId(any(String.class))).thenReturn(Optional.ofNullable(userProfile));
+        when(idamFeignClientMock.addUserRoles(updateUserProfileData.getRolesAdd(), "1234")).thenReturn(responseMock);
+
+        UserProfileRolesResponse response = sut.updateRoles(updateUserProfileData, userProfile.getIdamId());
+        assertThat(response.getRoleAdditionResponse().getIdamStatusCode()).isEqualTo("500");
+
+        verify(userProfileRepositoryMock, times(1)).findByIdamId(any(String.class));
+        verify(idamFeignClientMock, times(1)).addUserRoles(any(), any(String.class));
+        verify(responseMock, times(1)).close();
+    }
+
+    @Test
+    public void addRoles_FeignException() {
+        RoleName roleName1 = new RoleName("pui-case-manager");
```

Comment:
prefix test for method

Reply:
This has been addressed in the following commit: https://github.com/hmcts/rd-user-profile-api/pull/385/commits/04d04c3144cda8c41c2c8b875f3a62b0fa855501